src/Sarif/AnalyzeContextBase.cs

```diff
@@ -16,42 +19,254 @@ public AnalyzeContextBase()
         // All of these properties are persisted to configuration XML and can be
         // passed using that mechanism. All command-line arguments are 
         // candidates to follow this pattern.
-        public IEnumerable<IOption> GetOptions()
+        public virtual IEnumerable<IOption> GetOptions()
```

Comment:
>GetOptions [](http://example.com/codeflow?start=44&length=10)

This is the way that a class declares it has serializable properties. This array is not complete. 

An 'export-config' command will use MEF to retrieve all instances of `IOptionsProvider` and subsequently dump all published properties to an XML representation. What's serialized is what's expressed in the underlying static descriptors.